### PR TITLE
[TM] add spec for ImageStore

### PR DESCRIPTION
**Pull request by mitulsavani**
## Summary
This PR solves part of this issue: #24875

## Changelog

[General] [Added] - add TM spec for ImageStore

## Test Plan

Flow passes

### CODE REVIEW

File: Libraries/Image/NativeImageStore.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  +hasImageForTag: (
+    uri: string, 
+    callback: (hasImage: boolean) => void,
+  ) => void;
+  +removeImageForTag: (
+    uri: string,
+  ) => void;
+  +addImageFromBase64: (
+    base64ImageData: string,
+    succees: (uri: string) => void,
```
Comment (ericlewis) [inline Libraries/Image/NativeImageStore.js]: Small typo here, should be `success`

File: Libraries/Image/NativeImageStore.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    failure: (error: string) => void;
+  ) => void;
+  +getBase64ForTag: (
+    uri: string,
+    succees: (base64ImageData: string) => void,
```
Comment (ericlewis) [inline Libraries/Image/NativeImageStore.js]: Same here. `success`

File: Libraries/Image/NativeImageStore.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    failure: (error: string) => void,
+  ) => void;
+}
+
+export default TurboModuleRegistry.getEnforcing<Spec>('ImageStoringManager');
```
Comment (ericlewis) [inline Libraries/Image/NativeImageStore.js]: This should be `ImageStoreManager`

File: Libraries/Image/NativeImageStore.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    uri: string, 
+    callback: (hasImage: boolean) => void,
+  ) => void;
+  +removeImageForTag: (
+    uri: string,
+  ) => void;
+  +addImageFromBase64: (
+    base64ImageData: string,
+    succees: (uri: string) => void,
+    failure: (error: string) => void;
```
Comment (ericlewis) [inline Libraries/Image/NativeImageStore.js]: I don’t think failure callback’s pass back a string arg. I think they are ‘Object’ types.

File: Libraries/Image/NativeImageStore.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  ) => void;
+  +getBase64ForTag: (
+    uri: string,
+    succees: (base64ImageData: string) => void,
+    failure: (error: string) => void,
```
Comment (ericlewis) [inline Libraries/Image/NativeImageStore.js]: Same as above

File: Libraries/Image/NativeImageStore.js (hunk 4 above)
Comment (mitulsavani) [reply Libraries/Image/NativeImageStore.js]: Would it then look something like this?
```
failure: Object;
```
OR
```
failure: (error: Object) => void;
```

**Comment by mitulsavani** [comment]
> Great work! Just a few minor changes. I also suggest that you group the iOS only methods and comment that they are iOS only

What I have found is `getBase64ForTag` method is present in both iOS and Android. How should I comment particularly that method? 

PS: Please correct me if I am wrong here
> Example
```
// iOS only
+hasImageForTag: (
    uri: string, 
    callback: (hasImage: boolean) => void,
  ) => void;
  +removeImageForTag: (
    uri: string,
  ) => void;
  +addImageFromBase64: (
    base64ImageData: string,
    success: (uri: string) => void,
    failure: (error: string) => void;
  ) => void;

// iOS and Android 
  +getBase64ForTag: (
    uri: string,
    success: (base64ImageData: string) => void,
    failure: (error: string) => void,
  ) => void;
```

**Comment by ericlewis** [comment]
@mitulsavani switch the common one to be at the top & remove its comment basically. We also need to address all of these test failures.

**Comment by fkgozali** [comment]
@mitulsavani are you going to make some updates to this PR some time today? If not, I can just import this and update the code per @ericlewis comments accordingly.

File: Libraries/Image/NativeImageStore.js (hunk 4 above)
Comment (fkgozali) [reply Libraries/Image/NativeImageStore.js]: `failure: (error: Object) => void,`

**Comment by fkgozali** [comment]
Actually, since the changes are minor, I'll just edit them when importing, so we can save some time :). We're almost done!

**Comment by mitulsavani** [comment]
@fkgozali yes I will definitely make changes by tonight!

**Pull request closed without merging**